browser/resources/settings/getting_started_page/brave_account_common.ts

```diff
@@ -0,0 +1,23 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+export function onEyeIconClicked(event: Event) {
+  event.preventDefault()
+  let type = 'password'
+  const target = event.target as Element
+  target.setAttribute('name',
+    target.getAttribute('name')!.replace(/[^-]*$/, match => {
```

Comment:
fwiw Sonarr is giving a warning about catastrophic backtracking here - as the intention is just to toggle between eye-on and eye-off maybe we could do something simpler?


```suggestion
    target.getAttribute('name') === 'eye-on' ? 'eye-off' : 'eye-on')
```